### PR TITLE
Allow users to request identity enhancements from providers

### PR DESCRIPTION
**Pull request by smangelsdorf**
I'm not 100% sold on 6ab7f82 so happy to peel that commit off if it's bad.

### CODE REVIEW

File: app/views/providers/index.html.erb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-<%= page_header('Providers') %>
+p%= page_header('Providers') %>
```
Comment (bradleybeddoes) [inline app/views/providers/index.html.erb]: Looks like a wayward edit here which results in a heading bug per: https://www.dropbox.com/s/36y5jf3q5q73ga3/Screenshot%202015-04-22%2009.06.25.png?dl=0

**Comment by bradleybeddoes** [comment]
I'm happy with the commit that introduces the outstanding number of enhancement requests as a label on the button.

Makes it quite prominent that there is an action(s) required to be taken.

**Comment by bradleybeddoes** [comment]
I note that when I click on requests and through to a particular user that after I click the dismiss button the request goes away, as expected.

However if I click enhance and then go and modify the subject the initial enhancement request remains, that is it must be explicitly dismissed which isn't indicated and seems a bit unintuitive.

How do you feel about dismissing the request when the 'Enhance Identity' is clicked? Perhaps this would also lead to the need for a UI to list old requests?

**Comment by smangelsdorf** [comment]
I tried to avoid automatic, magic behaviour. In particular, I don't want things to disappear if somebody closes a browser tab part way through.

The flow I had in mind here was:

1. Request appears
2. Provider admin views request, clicks 'Enhance Identity'
3. Two options are available, so the provider admin emails the user asking for clarification on exactly what level of access they needed
4. Provider admin gets distracted while waiting for a reply and closes the browser tab

I guess I could modify the flow so that the request is dismissed when an attribute is provided via the "Enhance Identity" link. In that case, the redirect would need to go elsewhere, as going back to view the request doesn't really make sense anymore..

File: app/views/providers/index.html.erb (hunk 1 above)
Comment (smangelsdorf) [inline app/views/providers/index.html.erb]: Fixed. Well spotted.

**Comment by bradleybeddoes** [comment]
Yeah I figured that was the type of flow you where trying to protect however we need to do something otherwise I fear these requests will just build up over time and never be dismissed.

I note the dismissed action also sets the subject that undertook the dismissal per https://github.com/ausaccessfed/identity-enhancement-service/pull/76/files#diff-c7a9af97058bde13b9eab8ce23f106e8R47 this isn't occurring when an admin enhances an identity and then doesn't bother to also dismiss.

You're idea is likely the only practical solution here. Perhaps in that case the view would still return to detail enhancements of the subjects identity as it does now but append the fact that request X was completed at the same time?

**Comment by smangelsdorf** [comment]
Alright. Get that into ya.

**Comment by bradleybeddoes** [comment]
no one tricks the rainbow cop

**Comment by bradleybeddoes** [comment]
resources

**Pull request merged**